src/common/LocationComparator.java

```diff
@@ -0,0 +1,12 @@
+package common;
+
+import java.util.Comparator;
+
+public class LocationComparator implements Comparator<Location> {
+    public int compare( Location l1, Location l2 ) {
+        double l1Cost = l1.getDistance() / l1.getPassability();
+        double l2Cost = l2.getDistance() / l2.getPassability();
```

Comment:
This needs to change I think. It will lead to us almost never traversing tiles with lower passability even when they are the only way to reach the goal. Should probably change this to something like l1.getDistance() + SOME_CONSTANT / l1.getPassability()?

Reply:
Additionally, is the location comparer going to be ran against every combination of directions? 

Adding a constant would be a good idea, having it try I think at least 5-10 squares in either direction to search for a valid point forward would be good I think.

Reply:
It basically gets run every time we insert a location in the min heap.
So the idea is that any unit wanting to use pathfinding would call planner.getNextDirection( final destination ) and this will add the adjacent locations to the min heap with the first element being the one with the least cost produced by the above comparator.

So retrieving that would be O(1) and adding things to the list would be O(logn) I think and the same for removing the root.

Worst case this would be 8 adjacent locations. And the above code would be run each time we insert into the non empty min heap.

But we just grab the best move from this heap and if we can go that direction that is what we return.

We can try and extend this to locations beyond the adjacent ones, but that would require modifying this code a little bit.